packages/shared/lpc/tests/legacy_order.test.ts

```
@@ -0,0 +1,125 @@
+  });
+
+  test('derived ordering matches legacy ordering for every slot', () => {
+    const entries = buildEntries(legacyCatalog);
+    const result = buildLpcCatalog({ entries });
+    mismatchedSlots = [];
+
+    for (const legacySlot of legacyCatalog.slots) {
+      const derivedSlot = result.slots.find((s) => s.slot === legacySlot.slot);
+      expect(derivedSlot).toBeDefined();
+
+      const derivedAssetIds = derivedSlot!.variants.map((v) => v.assetId);
```

Biome (2.5.7)

[error] Forbidden non-null assertion.

(lint/style/noNonNullAssertion)